libs/flags/json_flag.go

```diff
@@ -57,6 +57,14 @@ func (j *JsonFlag) Unmarshal(v any) diag.Diagnostics {
 		return diags
 	}
 
+	if !nv.IsValid() {
+		return diags.Append(diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Invalid command input",
+			Detail:   fmt.Sprintf("expected input of type %s, received %s", reflect.TypeOf(v), dv.Kind()),
```

Comment:
The more important difference to highlight is that we were expecting a JSON, the type itself does not mean anything to a CLI user.

```suggestion
			Detail:   fmt.Sprintf("expected JSON input of type %s, received %s", reflect.TypeOf(v), dv.Kind()),
```

Reply:
They still can provide an incorrect type, for example, plain string vs. map and etc. But worth mentioning JSON indeed

Reply:
Replaced SDK types with JSON types here: https://github.com/databricks/cli/pull/3398/commits/f9192cab24fc513dcdf09de517096007a2517074